interpreter/valid/valid.ml

```diff
@@ -260,20 +264,20 @@ let rec check_instr (c : context) (e : instr) (s : infer_stack_type) : op_type =
     [t] --> []
 
   | Load memop ->
-    check_memop c memop (Lib.Option.map fst) e.at;
-    [I32Type] --> [memop.ty]
+    let t = check_memop c memop (Lib.Option.map fst) e.at in
+    [t] --> [memop.ty]
```

Comment:
Nit: move the conversion from index to value_type out of check_memop to here, in order to appear symmetric to the MemoryInstr typing rules below.